Source/VirgilCrypto/VirgilCrypto+Impl.swift

```diff
@@ -0,0 +1,267 @@
+            result = cipher.packMessageInfo()
+
+            result! += try cipher.processEncryption(data: inputData)
+
+            result! += try cipher.finishEncryption()
+
+            if let signingOpt = signingOptions, signingOpt.mode == .signThenEncrypt {
+                result! += try cipher.packMessageInfoFooter()
+            }
+
+        case let .stream(inputStream, streamSize, outputStream):
+            if inputStream.streamStatus == .notOpen {
+                inputStream.open()
+            }
+            if outputStream.streamStatus == .notOpen {
+                outputStream.open()
+            }
+
+            try StreamUtils.write(cipher.packMessageInfo(), to: outputStream)
+
+            try StreamUtils.forEachChunk(in: inputStream, streamSize: streamSize) {
+                try StreamUtils.write(try cipher.processEncryption(data: $0), to: outputStream)
+            }
+
+            try StreamUtils.write(try cipher.finishEncryption(), to: outputStream)
+
+            if let signingOpt = signingOptions, signingOpt.mode == .signThenEncrypt {
+                try StreamUtils.write(try cipher.packMessageInfoFooter(), to: outputStream)
+            }
+        }
+
+        return result
+    }
+
+    internal func decrypt(inputOutput: InputOutput,
+                          verifyingOptions: VerifyingOptions?,
+                          privateKey: VirgilPrivateKey) throws -> Data? {
+        let cipher = RecipientCipher()
+        cipher.setRandom(random: self.rng)
+
+        try cipher.startDecryptionWithKey(recipientId: privateKey.identifier,
+                                          privateKey: privateKey.key,
+                                          messageInfo: Data())
+
+        var result: Data? = nil
+
+        switch inputOutput {
+        case let .stream(inputStream, _, outputStream):
+            if inputStream.streamStatus == .notOpen {
+                inputStream.open()
+            }
+            if outputStream.streamStatus == .notOpen {
+                outputStream.open()
+            }
+
+            try StreamUtils.forEachChunk(in: inputStream, streamSize: nil) {
+                try StreamUtils.write(try cipher.processDecryption(data: $0), to: outputStream)
+            }
+            try StreamUtils.write(try cipher.finishDecryption(), to: outputStream)
+
+        case let .data(input):
+            result = Data()
+
+            result! += try cipher.processDecryption(data: input)
+
+            result! += try cipher.finishDecryption()
+        }
+
+        if let verifyingOpt = verifyingOptions {
+            var mode = verifyingOpt.mode
+
+            if mode == .any {
+                mode = cipher.isDataSigned() ? .decryptThenVerify : .decryptAndVerify
+            }
+
+            switch mode {
+            case .decryptAndVerify:
+                guard case InputOutput.data(_) = inputOutput else {
+                    fatalError("signAndEncrypt is not supported for streams")
```

Comment:
signAndDecrypt*

Reply:
signAndDecrypt? what's that?

Reply:
Maybe it should be "decryptAndVerify"  in fatal error message, since it is decrypt func? I also misspelled it at first comment.